### PR TITLE
feat(accordion): ionChange will only emit from user committed changes

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
  - Some docs updates need to be made in the `ionic-docs` repo, in a separate PR. See the [contributing guide](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#modifying-documentation) for details.
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue URL: N/A Internal ticket

## What is the new behavior?

- `ionChange` will now fire as a result of user-generated actions only.
- The one exception is when an app passes an array of values to a single-select accordion. When that happens, we set the `value` to be the first item in the array and re-emit `ionChange`.

## Does this introduce a breaking change?

- [x] Yes
- [ ] No

`ionChange` will no longer be emitted when the `value` of `ion-accordion-group` is modified externally. `ionChange` will only be emitted from user committed changes, such as clicking or tapping the accordion header.

### CODE REVIEW

File: angular/src/directives/proxies.ts [hunk 1]
```diff
@@ -31,7 +31,10 @@ export class IonAccordion {
 import type { AccordionGroupChangeEventDetail as IAccordionGroupAccordionGroupChangeEventDetail } from '@ionic/core';
 export declare interface IonAccordionGroup extends Components.IonAccordionGroup {
   /**
-   * Emitted when the value property has changed. 
+   * Emitted when the value property has changed
```
Comment (liamdebeasi) [inline angular/src/directives/proxies.ts:34]: This was auto-generated. The formatting looks weird, but should be fine?

Comment (sean-perkins) [reply angular/src/directives/proxies.ts:34]: Yeah this should be fine. Intellisense still formats the text correctly. Just a quirk of the output target generator (we would have to split on line breaks and generate per line - which is possible, but the raw output works as is for now).

**Review by sean-perkins — APPROVED** [review]
Good work here 👍

**Comment by amandaejohnston** [comment]
Holding on review until tests are passing; seeing a bunch of random failures 🤔 Not sure what's going on there.

**Comment by liamdebeasi** [comment]
Good to review now. There were some issues with installing dependencies for Playwright. Possibly related to https://www.githubstatus.com/incidents/8blgv351d71j.

**Review by amandaejohnston — APPROVED** [review]

**Pull request merged**